File: test/bot_sources/perl/basic_bot.pl
#!/usr/bin/env perl

while (<>) {
    chomp $_;
    @_ = split(' ', $_);

    if (@_[0] eq "settings") {
        print STDERR "@_\n";
    } elsif (@_[0] eq "update") {
        print STDERR "@_\n";
    } elsif (@_[0] eq "action") {
        $random_number = int(rand(3));

        if ($random_number == 0) {
            print "rock\n";
        } elsif ($random_number == 1) {
            print "paper\n";
        } elsif ($random_number == 2) {
            print "scissors\n";
        }
    }
}

File: test/bot_sources/c/basic_bot.c
#define _CRT_SECURE_NO_WARNINGS

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>


int main(void) {
    char    line[16384];

    while(fgets(line, 16384, stdin) != NULL) {
        if (strncmp(line, "action", 6) == 0) {
            int randNumb = rand() % 3;
            if(randNumb == 0) {
                fprintf(stdout, "rock\n");
            }
            if(randNumb == 1) {
                fprintf(stdout, "paper\n");
            }
            if(randNumb == 2) {
                fprintf(stdout, "scissors\n");
            }
			fflush(stdout);
            continue;
        }
    }

    return 0;
}
